test/unit/org/apache/cassandra/service/LeaveAndBootstrapTest.java

```diff
@@ -88,6 +90,7 @@ public void newTestWriteEndpointsDuringLeave() throws Exception
         TokenMetadata tmd = ss.getTokenMetadata();
         tmd.clearUnsafe();
         Gossiper.instance.clearUnsafe();
+        Nodes.peers().get().forEach(peer -> Nodes.peers().remove(peer.getPeerAddressAndPort(), true, true));
```

Comment:
Maybe Gossiper.instance.clearUnsafe() should do this?

Reply:
yes, I think it can